calyx-backend/src/verilog.rs

```diff
@@ -110,23 +193,88 @@ impl Backend for VerilogBackend {
         ctx: &ir::Context,
         file: &mut OutputFile,
     ) -> CalyxResult<()> {
+        // Declare all library configurations here
+        let libraries = vec![LibraryConfig::default()];
+
+        let mut all_imported_files: Vec<String> = collect_imported_files(ctx);
+
+        let mut morty_config = MortyConfig::default();
+        // Check if any special library is needed and perform some pre-processing by invoking their corresponding handlers
+        let mut library_needed = false;
+        for lib in &libraries {
+            if (lib.detect_handler)(ctx) {
+                (lib.library_handler)(
+                    &mut morty_config,
+                    &mut all_imported_files,
+                )?;
+                library_needed = true;
+            }
+        }
+
         let fw = &mut file.get_write();
-        for extern_path in &ctx.lib.extern_paths() {
-            // The extern file is guaranteed to exist by the frontend.
-            let mut ext = File::open(extern_path).unwrap();
-            io::copy(&mut ext, fw).map_err(|err| {
-                let std::io::Error { .. } = err;
-                Error::write_error(format!(
-                    "File not found: {}",
-                    file.as_path_string()
-                ))
-            })?;
-            // Add a newline after appending a library file
-            writeln!(fw)?;
+        // If we needed a special library (like HardFloat), run Morty to pickle the files
+        if library_needed {
+            let final_data = Value::Array(vec![morty_config.to_json()]);
+
+            let mut morty = OpenOptions::new()
+                .write(true)
+                .create(true)
+                .truncate(true)
+                .open("target/tmp/morty.json")?;
+
+            writeln!(morty, "{}", serde_json::to_string_pretty(&final_data)?)?;
+            morty.flush()?;
+
+            // Invoke Morty
+            let cmd = Command::new("morty")
```

Comment:
Have you considered including `morty` as a Rust crate instead of launching it as an external binary? With a crate, then the user would not have to install it.

Reply:
`morty` is available on crates.io: https://crates.io/crates/morty

I have not investigated the API though.